gatorgrade/generate/generate.py

```diff
@@ -1,21 +1,82 @@
-"""Generate a GatorGrade configuration file.
+"""Generate a YAML file with default messages and specific paths."""
+import os
+from typing import List
 
-Gatorgrade file will include paths to whitelisted files
-and default GatorGrader checks.
-"""
+# Define colors for terminal output
+OKGREEN = "\033[92m"
+WARNING = "\033[93m"
+FAIL = "\033[91m"
 
-# Import the necessary libraries
 
-
-def create_targeted_paths_list():
+def create_targeted_paths_list(
+    key_word_list: List[str], relative_run_path: str = "."
+) -> List[str]:
     """Generate a list of targeted paths by walking the paths."""
+    targeted_paths = []
     # Go through the root repo, the sub dictionaries and files.
-    # Select only files in the dictionaries with specific names.
-    # Add those targeted file paths into a list and returns it.
+    # The os.walk will only scan the paths.
+    # So the empty folders containing nothing won't be gone through
+    for dirpath, _, filenames in os.walk(relative_run_path):
+        # Split path string into multiple layers of directories
+        path_dir_list = dirpath.split("/")
+        # Ignore folder starting with double underscore
+        if any(path.startswith("__") for path in path_dir_list):
+            continue
+        # Ignore hidden folders and first layer. the root repo is always dot
+        # Keep double dot. It represents the returning sign
+        if any(
+            path.startswith(".") and not path.startswith("..")
+            for path in path_dir_list[1:]
+        ):
+            continue
+        for filename in filenames:
+            # Ignore the file starting with double underscore and hidden file
+            if filename.startswith("__") or filename.startswith("."):
+                continue
+            # Add paths when they have the key words in the second and the third directories
+            # For the path with only two directories,
+            # check key words in the second directory folder name
+            if len(path_dir_list) == 2:
+                if path_dir_list[1] in key_word_list:
+                    targeted_paths.append(os.path.join(dirpath, filename))
+
+            # For the other paths with more than 2 directories,
+            # check key words in the second and third directories
```

Comment:
Why does it only look 2 or 3 level deep? Maybe there isn't a use case for 3+, but if it would be helpful to go deeper, I think generalizing the code that you have a bit would also help make it easier to follow!

It might not be worth the optimization yet though if there's no current need to go deeper than 3 directories.

Reply:
Hey, I am willing to explain it. @mariakimheinert feels it's better just to cover 3 levels deep to avoid the setting up folders may contain some files or folders sharing the same names with targeted ones. However, I just heard that @mariakimheinert may expect some change in input interface, so this part may change